### PR TITLE
parser: support pre-split global index add special comment support for pre_split index option

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/57552, ref https://github.com/pingcap/tidb/issues/57551

Problem Summary:

Wrap the pre_split index option with the special comment to avoid TiCDC failing to sync downstream.

### What changed and how does it work?

See code changes.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/ast/ddl.go [hunk 1]
```diff
@@ -828,17 +828,20 @@ func (n *IndexOption) Restore(ctx *format.RestoreCtx) error {
...
 		}
-		ctx.WriteKeyWord("PRE_SPLIT_REGIONS")
-		ctx.WritePlain(" = ")
-		if n.SplitOpt.Num != 0 && len(n.SplitOpt.Lower) == 0 {
-			ctx.WritePlainf("%d", n.SplitOpt.Num)
-		} else {
-			ctx.WritePlain("(")
-			if err := n.SplitOpt.Restore(ctx); err != nil {
-				return errors.Annotate(err, "An error occurred while splicing IndexOption SplitOpt")
+		_ = ctx.WriteWithSpecialComments(tidb.FeatureIDPresplit, func() error {
```
Comment (D3Hunter) [inline pkg/parser/ast/ddl.go:831]:
```suggestion
		return ctx.WriteWithSpecialComments(tidb.FeatureIDPresplit, func() error {
```
and remove the trailing `return nil`

File: tests/realtikvtest/addindextest3/functional_test.go [hunk 2]
```diff
@@ -213,6 +213,15 @@ func TestAddIndexPresplitIndexRegions(t *testing.T) {
...
 	tk.MustExec("drop index idx on t;")
 	tk.MustExec("set @@global.tidb_ddl_enable_fast_reorg = off;")
+
+	tk.MustExec("drop table t;")
+	tk.MustExec("set @@global.tidb_ddl_enable_fast_reorg = on;")
+	tk.MustExec("set @@global.tidb_enable_dist_task = off;")
+	tk.MustExec("create table t (a int, b int) partition by range (b)" +
+		" (partition p0 values less than (10), " +
+		"  partition p1 values less than (maxvalue));")
+	tk.MustExec("alter table t add unique index p_a (a) global pre_split_regions = (by (5), (15));")
```
Comment (D3Hunter) [inline tests/realtikvtest/addindextest3/functional_test.go:223]: maybe use comments to specify the option to check that it works

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: tests/realtikvtest/addindextest3/functional_test.go (hunk 2 above)
Comment (tangenta) [reply tests/realtikvtest/addindextest3/functional_test.go:240]: https://github.com/pingcap/tidb/pull/58408/files#diff-097dd78eea6150a40af20fc573492ca816b0dd01a05d0c2f4174511162c050c0R196-R198

**Review by D3Hunter — APPROVED** [review]

File: pkg/ddl/index_presplit.go [hunk 3]
```diff
@@ -120,6 +118,17 @@ func getSplitIdxKeysFromValueList(
 		return getSplitIdxPhysicalKeysFromValueList(sctx, tblInfo, idxInfo, tblInfo.ID, byRows, destKeys)
 	}
 
+	if idxInfo.Global {
+		var idxTblID int64
+		if pi.NewTableID != 0 && idxInfo.State != model.StatePublic {
```
Comment (wjhuang2016) [inline pkg/ddl/index_presplit.go:123]: Please add comment for this

Comment (tangenta) [reply pkg/ddl/index_presplit.go:123]: This is used in reorg partition. Since we don't support pre_split option in reorg partition, let's remove this.

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**